# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1539915067 14400
#      Thu Oct 18 22:11:07 2018 -0400
# Node ID 6882ea1ce4ab04dc4b48492250314d4e194f84d4
# Parent  9e791a62bcbaca1898240d11d298c9b1bae9e178
allow access to C-style argc and argv from application class

* octave.h (cmdline_options::sys_argc, cmdline_options::sys_argv,
application::sys_argc, application::sys_argv): New functions.

diff --git a/libinterp/octave.h b/libinterp/octave.h
--- a/libinterp/octave.h
+++ b/libinterp/octave.h
@@ -43,16 +43,19 @@ namespace octave
     cmdline_options (void);
 
     cmdline_options (int argc, char **argv);
 
     cmdline_options (const cmdline_options&) = default;
 
     cmdline_options& operator = (const cmdline_options&) = default;
 
+    int sys_argc (void) const { return m_all_args.numel (); }
+    char **sys_argv (void) const { return m_all_args.c_str_vec (); }
+
     bool debug_jit (void) const { return m_debug_jit; }
     bool echo_commands (void) const { return m_echo_commands; }
 
     OCTAVE_DEPRECATED (4.4, "use 'gui' instead")
     bool force_gui (void) const { return m_gui; }
 
     bool forced_interactive (void) const { return m_forced_interactive; }
     bool forced_line_editing (void) const { return m_forced_line_editing; }
@@ -250,16 +253,19 @@ namespace octave
     // No copying, at least not yet...
 
     application (const application&) = delete;
 
     application& operator = (const application&) = delete;
 
     virtual ~application (void);
 
+    int sys_argc (void) const { return m_options.sys_argc (); }
+    char **sys_argv (void) const { return m_options.sys_argv (); }
+
     void set_program_names (const std::string& pname);
 
     void intern_argv (const string_vector& args);
 
     cmdline_options options (void) const { return m_options; }
 
     bool have_eval_option_code (void) const { return m_have_eval_option_code; }
 
